src/chain.cljs

```diff
@@ -0,0 +1,200 @@
+  (hash [this] (m/hash coll))
+  (get [this k not-found] (_ (m/get coll k not-found)))
+  (hasKey [this key] (m/hasKey coll key))
+  (reverse [this] (_ (m/reverse coll)))
+  (take [this] (_ (m/take coll)))
+  (take [this n] (_ (m/take n coll)))
+  (drop [this] (_ (m/drop coll)))
+  (drop [this n] (_ (m/drop n coll)))
+  (takeNth [this n] (_ (m/takeNth n coll)))
+  (partition [this n] (_ (m/partition n coll)))
+  (partition [this n step] (_ (m/partition n step coll)))
+  (partition [this n step pad] (_ (m/partition n step pad coll)))
+  (partitionAll [this n] (_ (m/partitionAll n coll)))
+  (partitionAll [this n step] (_ (m/partitionAll n step coll)))
+  (partitionBy [this f] (_ (m/partitionBy f coll)))
+  (iterate [this f] (_ (m/iterate coll f coll)))
+  (into [this from] (_ (m/into coll from)))
+  (intoArray [this] (_ (m/intoArray coll)))
+  (merge [this & maps] (_ (apply m/merge coll maps)))
+  ;; TODO: Fix apply issue here
+  (mergeWith [this f & maps] (_ (apply m/merge f coll maps)))
+  (subvec [this start end] (_ (m/subvec coll start end)))
+  (takeWhile [this pred] (_ (m/takeWhile pred coll)))
+  (dropWhile [this pred] (_ (m/dropWhile pred coll)))
+  (groupBy [this f] (_ (m/groupBy f coll)))
+  (interpose [this sep] (_ (m/interpose sep coll)))
+  (interleave [this & colls] (_ (apply m/interleave coll colls)))
+  (concat [this & zs] (_ (apply m/concat coll zs)))
+  (sort [this] (_ (m/sort coll)))
+  (sortBy [this f] (_ (m/sortBy f coll)))
+  (vec [this] (_ (vec coll)))
+  
+  ;; Should prStr be here for clj programmers?
+  (prStr [this] (pr-str coll))
+  (isEqual [this other] (= coll
+                           ;; Compare against a potentiall wrapped/unwrapped other
+                           (or (.-coll other)
+                               other)))
+  ;; Same as isEqual, apparently more idiomatic with ES6 now
+  (is [this other] (= coll (or (.-coll other)
+                               other)))
+  (isEmpty [this] (m/isEmpty coll))
+
+  ;; It may be impossible to create a list through this interface, but
+  ;; useful in case a manually created one is passed in
+  (isList [this] (m/isList coll))
+  (isSeq [this] (m/isSeq coll))
+  (isVector [this] (m/isVector coll))
+  (isMap [this] (m/isMap coll))
+  (isSet [this] (m/isSet coll))
+  
+  (isSymbol [this] (m/isSymbol coll))
+  
+  (isCollection [this] (m/isCollection coll))
+  (isSequential [this] (m/isSequential coll))
+  (isAssociative [this] (m/isAssociative coll))
+  (isCounted [this] (m/isCounted coll))
+  (isIndexed [this] (m/isIndexed coll))
+  (isReduceable [this] (m/isReduceable coll))
+  (isSeqable [this] (m/isSeqable coll))
+  (isReversible [this] (m/isReversible coll))
+
+  (isChainable [_] true)
+  (toSeq [this] (_ (m/seq coll)))
+  (toSet [this] (_ (m/set coll)))
+  ;; Not sure if .toJs or .out makes more sense for the JS world. Perhaps .out should give the .-coll out so other mori functions could use it, and .toJs should be for converting to JS values?
```

Comment:
Highlighting `.out` vs `.toJS` here.

`.out` was supposed to be analogous to underscore's `.val`, and as a nice way of hiding the fact that the computations aren't happening on vanilla JS objects. I think it might be better to have `.out` return the `coll` of the Chainable, so that it can be used with other mori functions, and to have `.toJs` be used exclusively for pulling values out at the end of a chain of computation.